src/app/notebook/notebook.component.ts

```diff
@@ -101,4 +103,31 @@ export class NotebookComponent implements OnChanges {
   isArray(value: any): boolean {
     return Array.isArray(value);
   }
+
+  cellsToNbformat4(cells: Cell[]): Cell[] {
+    if (this.isArray(cells)) {
+      cells.forEach((cell) => this.cellToNbFormat4(cell));
+    }
+    return cells;
+  }
+
+  cellToNbFormat4(cell: Cell): void {
+    if (cell.cell_type === 'code') {
+      // Handle some field name changes: 'input' -> 'source', 'prompt_number' -> 'execution_count'
+      cell.source ??= cell['input'];
+      cell.execution_count ??= cell['prompt_number'];
+      // Convert each output.
+      if (this.isArray(cell.outputs)) {
+        cell.outputs.forEach(
+          (output) =>
+            // Map the old representation of rich outputs to a "mime bundle'.
+            (output.data ??= {
+              'text/plain': output['text'],
```

Comment:
Is it ok for `output['text']` (or the other ones) to be null?

If it's OK, you should add null to the MimeBundle definition:

```
export interface MimeBundle {
  [mimeType: string]: string | string[] | null;
}
```